nautobot_golden_config/views.py

```diff
@@ -824,24 +920,47 @@ class ConfigPlanUIViewSet(NautobotUIViewSet):
     lookup_field = "pk"
     action_buttons = ("add",)
 
+    def alter_queryset(self, request):
+        """Build actual runtime queryset as the build time queryset provides no information."""
+        if "Completed" in request.query_params.get("status", []):
+            return models.ConfigPlan.objects.all()
+
+        return models.ConfigPlan.objects.exclude(status__name="Completed")
+
     def get_form_class(self, **kwargs):
         """Helper function to get form_class for different views."""
         if self.action == "update":
             return forms.ConfigPlanUpdateForm
         return super().get_form_class(**kwargs)
 
-    def create(self, request, *args, **kwargs):
-        """Helper function to warn if the Job is not enabled to run."""
-        job = Job.objects.get(name="Generate Config Plans")
-        if not job.enabled:
-            messages.warning(
+    def get_extra_context(self, request, instance=None):
+        """A ConfigPlan helper function to warn if the Job is not enabled to run."""
+        jobs = []
+        jobs.append(
+            [
+                Job.objects.get(module_name="nautobot_golden_config.jobs", job_class_name="GenerateConfigPlans"),
+                request,
```

Comment:
is it possible to use string-ified imported elements here?

Reply:
I am not sure about the request or the benefits, do you mind explaining further?

Reply:
I was thinking we could avoid re-typing class names, possibly making it harder to make mistake in the future.